src/main/java/filter/expression/QualifierContent.java

```diff
@@ -0,0 +1,66 @@
+package filter.expression;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class QualifierContent {
+
+    //Only one of these will exist in a QualifierContent object
```

Comment:
Is there a more elegant way to handle this 'only one can be present' condition? e.g. inheritance, enums